core/src/test/java/com/google/cloud/sql/core/CloudSqlInstanceTest.java

```diff
@@ -111,12 +111,49 @@ public void testInstanceFailsOnConnectionError() throws Exception {
             stubCredentialFactory,
             executorService,
             keyPairFuture,
-            RateLimiter.create(1.0 / 30.0));
+            RateLimiter.create(10));
 
     RuntimeException ex = Assert.assertThrows(RuntimeException.class, instance::getSslData);
     assertThat(ex).hasMessageThat().contains("always fails");
   }
 
+  @Test
+  public void testInstanceFailsOnTooLongToRetrieve() throws Exception {
+
+    InstanceDataSupplier instanceDataSupplier =
+        (CloudSqlInstanceName instanceName,
+            AccessTokenSupplier accessTokenSupplier,
+            AuthType authType,
+            ListeningScheduledExecutorService exec,
+            ListenableFuture<KeyPair> keyPair) -> {
+          ListenableFuture<?> f =
+              exec.submit(
+                  () -> {
+                    try {
+                      Thread.sleep(120 * 1000); // stop for 2 minutes
+                    } catch (InterruptedException e) {
+                    }
+                    throw new RuntimeException("fake read timeout");
+                  });
+          f.get(); // this will throw an ExecutionException
```

Comment:
Let's add some more details to the comment -- it will throw an exception because we've configured the connector wait only 30s?

Reply:
I added comments and cleaned this up.